### PR TITLE
Feature/l3 198

### PR DESCRIPTION
**Pull request by Ellenn-A**
# Refactor NPM package handling 

- added a folder for handling of packages with npm => new class: `NPMPackageHandler`
- removed 'getFiles' folder/class - no longer needed 
- updated test in `checkVersion.test.ts` & added additional test to check functionality of new `NPMPackageHandler` class

# When reviewing please look at: 
- `checkVersion.test.ts` comment on line `34` not sure if we would want to assert for this

### CODE REVIEW

File: src/lib/npm/index.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import fsPromises from 'fs/promises'
+import { z } from 'zod'
+
+
+const packageParser = z.object({
+    version: z.string()
+})
+
+export default class NPMPackageHandler {
```
Comment (mattdean-digicatapult) [inline src/lib/npm/index.ts:9]: This class needs unit tests similar to what we've done for cargo. Also there's some random whitespace here that I'm surprised the linter is happy with. Might want to have a look at that

File: src/lib/checkVersion.ts [hunk 2]
```diff
@@ -96,29 +92,19 @@ export class CheckVersion {
...
+
+    if (!(versions['packageJsonLock']) || !(versions['packageJson'])){
+      this.core.setFailed(`No versions found for package or package-lock.`)
     }
   
-    this.compareVersions(packageJson['version'] || '', packageLockJson['version'] || '')
+    this.compareVersions(versions['packageJson'] || '', versions['packageJsonLock'] || '')
   
-    return packageLockJson['version'] 
+    return versions['packageJsonLock']
```
Comment (mattdean-digicatapult) [inline src/lib/checkVersion.ts:107]: I'm wondering if the comparison logic should also move to the npm class as this doens't apply for cargo. Thoughts?

File: .prettierignore [hunk 3]
```diff
@@ -1,3 +1,3 @@
 dist/
-lib/
+# lib/
```
Comment (mattdean-digicatapult) [inline .prettierignore:2]: can we just remove this line please. Can't believe that sneaked in

Comment (Ellenn-A) [reply .prettierignore:2]: Yeah imagine how confused I was when I kept trying to format on save and nothing was happening

**Review by mattdean-digicatapult — APPROVED** [review]

**Pull request merged**